taker-frontend/src/components/History.tsx

```diff
@@ -86,16 +76,7 @@ const CfdDetails = ({ cfd }: CfdDetailsProps) => {
     const txCet = cfd.details.tx_url_list.find((tx) => tx.label === TxLabel.Cet);
     const txSettled = cfd.details.tx_url_list.find((tx) => tx.label === TxLabel.Collaborative);
 
-    let { run: postAction, isLoading: isActioning } = useAsync({
-        deferFn: async ([orderId, action]: any[]) => {
-            try {
-                console.log(`Closing: ${orderId} ${action}`);
-                await doPostAction(orderId, action);
-            } catch (e) {
-                createErrorToast(toast, e);
-            }
-        },
-    });
+    let [settle, isSettling] = usePostRequest(`/api/cfd/${cfd.order_id}/settle`);
```

Comment:
this will need to become general really soon as @rishflab is working on using "commit" action if there's no maker online). Perhaps it's worth keeping it general with `postAction, isActioning` with the action name argument?

Reply:
You can just make the URL dynamic? That is easily supported :)